### PR TITLE
Enhanced RDoc for literals.rdoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Makes link targets among percent literals.
- Adds links to those targets.
- Adds examples to percent literals.
- Links from opening summary list to corresponding sections

### CODE REVIEW

File: doc/syntax/literals.rdoc [hunk 1]
```diff
@@ -299,6 +311,11 @@ Like strings, a single-quote may be used to disable interpolation:
 When creating a Hash, there is a special syntax for referencing a Symbol as
 well.
 
+See also:
+
+* {%r: Symbol Literals}[#label-25s-3A+Symbol+Literals]
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:316]: This looks wrong. `%r` is for regexp literals, not symbol literals.

File: doc/syntax/literals.rdoc [hunk 2]
```diff
@@ -383,25 +409,75 @@ This proc will add one to its argument.
...
+
+=== <tt>% and %Q</tt>: Interpolable String Literals
+
+You can write an interpolable string with <tt>%Q</tt>
+or with its alias <tt>%</tt>:
+
+  %[foo bar baz]       # => "foo bar baz"
+  %(1 + 1 is #{1 + 1}) # => "1 + 1 is 2" # Interpolation.
+
+=== <tt>%w</tt>: String-Array Literals
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:444]: You should probably discuss `%W` here  (which allows interpolation), or in a separate heading directly after.

File: doc/syntax/literals.rdoc [hunk 3]
```diff
@@ -383,25 +409,75 @@ This proc will add one to its argument.
...
+
+You can write an array of strings with <tt>%w</tt>:
+
+  %w[foo bar baz]       # => ["foo", "bar", "baz"]
+  %w[1 % *]             # => ["1", "%", "*"]
+  # Use backslash to embed spaces in the strings.
+  %w[foo\ bar baz\ bat] # => ["foo bar", "baz bat"]
+
+=== <tt>%i</tt>: Symbol-Array Literals
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:453]: You should probably discuss `%I` here (which allows interpolation), or in a separate heading directly after.

File: doc/syntax/literals.rdoc [hunk 4]
```diff
@@ -383,25 +409,75 @@ This proc will add one to its argument.
...
+=== <tt>%x</tt>: Backtick Literals
 
-  %w[one one-hundred\ one]
-  #=> ["one", "one-hundred one"]
+You can write and execute a shell command with <tt>%x</tt>:
 
-If you are using "(", "[", "{", "<" you must close it with ")", "]", "}", ">"
-respectively.  You may use most other non-alphanumeric characters for percent
-string delimiters such as "%", "|", "^", etc.
+    %x(time /t) # => "02:23 PM\n"
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:483]: Considering how few Ruby users will be running on Windows, we should probably avoid Windows-specific examples here.  Maybe `%x(echo 1) # => "1\n"`, which should be the same on Windows and non-Windows.

File: doc/syntax/literals.rdoc [hunk 5]
```diff
@@ -310,6 +327,11 @@ You may place expressions inside the array:
   [1, 1 + 1, 1 + 2]
   [1, [1 + 1, [1 + 2]]]
 
+See also:
+
+* {%w: String-Array Literals}[#label-25w-3A+String-Array+Literals]
+* {%i: Symbol-Array Literals}[#label-25i-3A+Symbol-Array+Literals]
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:333]: Should mention `%W` and `%I` here, for consistency with mentioning both `%q` and `%Q` above.

File: doc/syntax/literals.rdoc [hunk 6]
```diff
@@ -2,15 +2,31 @@
...
+* {Hash Literals}[#label-Hash+Literals]
+* {Range Literals}[#label-Range+Literals]
+* {Regexp Literals}[#label-Regexp+Literals]
+* {Lambda Proc Literals}[#label-Lambda+Proc+Literals]
+* {Percent Literals}[#label-Percent+Literals]
+
+  * {%q: Non-Interpolable String Literals}[#label-25q-3A+Non-Interpolable+String+Literals]
+  * {% and %Q: Interpolable String Literals}[#label-25+and+-25Q-3A+Interpolable+String+Literals]
+  * {%w: String-Array Literals}[#label-25w-3A+String-Array+Literals]
+  * {%i: Symbol-Array Literals}[#label-25i-3A+Symbol-Array+Literals]
```
Comment (jeremyevans) [inline doc/syntax/literals.rdoc:26]: `%W` and `%I` should also be discussed and linked from this section.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
This looks mostly good.  Please see the inline comments.

File: doc/syntax/literals.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/syntax/literals.rdoc]: Fixed.

File: doc/syntax/literals.rdoc (hunk 2 above)
Comment (BurdetteLamar) [reply doc/syntax/literals.rdoc]: Done.

File: doc/syntax/literals.rdoc (hunk 3 above)
Comment (BurdetteLamar) [reply doc/syntax/literals.rdoc]: Done.

File: doc/syntax/literals.rdoc (hunk 4 above)
Comment (BurdetteLamar) [reply doc/syntax/literals.rdoc]: Done.

File: doc/syntax/literals.rdoc (hunk 5 above)
Comment (BurdetteLamar) [reply doc/syntax/literals.rdoc]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**